internal/services/recoveryservices/site_recovery_replicated_vm_resource.go

```diff
@@ -314,61 +317,143 @@ func networkInterfaceResource() *pluginsdk.Resource {
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
 
-			"target_static_ip": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     false,
-				ValidateFunc: validation.StringIsNotEmpty,
-			},
+						"failover_test_static_ip": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
 
-			"failover_test_subnet_name": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				Computed:     true,
-				ForceNew:     false,
-				ValidateFunc: validation.StringIsNotEmpty,
-			},
+						"target_static_ip": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
 
-			"target_subnet_name": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ForceNew:     false,
-				ValidateFunc: validation.StringIsNotEmpty,
-			},
+						"failover_test_subnet_name": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
 
-			"failover_test_public_ip_address_id": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				Computed:     true,
-				ForceNew:     false,
-				ValidateFunc: azure.ValidateResourceID,
-			},
+						"target_subnet_name": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
 
-			"recovery_load_balancer_backend_address_pool_ids": {
-				Type:     pluginsdk.TypeSet,
-				Optional: true,
-				Elem: &pluginsdk.Schema{
-					Type:         pluginsdk.TypeString,
-					ValidateFunc: loadbalancers.ValidateLoadBalancerBackendAddressPoolID,
+						"failover_test_public_ip_address_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: azure.ValidateResourceID,
+						},
+
+						"recovery_load_balancer_backend_address_pool_ids": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: loadbalancers.ValidateLoadBalancerBackendAddressPoolID,
+							},
+						},
+
+						"recovery_public_ip_address_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: azure.ValidateResourceID,
+						},
+
+						"is_primary": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+					},
 				},
 			},
+		},
+	}
 
-			"recovery_public_ip_address_id": {
+	if !features.FivePointOh() {
+		nicSchema.Schema["ip_configuration"].Computed = true
+
+		nicSchema.Schema["failover_test_static_ip"] = &pluginsdk.Schema{
+			Deprecated:   "this property has been deprecated in favour of `network_interface.ip_configuration`",
```

Comment:
Could you update each of these deprecation messages to point to the exact replacement property, and to include that they will be removed in 5.0?

```suggestion
			Deprecated:   "this property has been deprecated in favour of `network_interface.ip_configuration.failover_test_static_ip` and will be removed in v5.0 of the AzureRM provider",
```